examples/LEVDisplay/LEVDisplay.ino

```diff
@@ -0,0 +1,207 @@
+void levAntChannelEvent(ChannelEventResponse& msg, uintptr_t data);
+
+void manufacturersInformationDataPageHandler(ManufacturersInformation& msg, uintptr_t data);
+void productInformationDataPageHandler(ProductInformation& msg, uintptr_t data);
+
+void printStatus(uint8_t status);
+
+void setup() {
+    Serial2.begin(BAUD_RATE);
+    ant.setSerial(Serial2);
+    delay(1000);
+
+    router.setDriver(&ant); // never touch ant again
+    router.setAntPlusNetworkKey(NETWORK_KEY);
+    router.setProfile(CHANNEL_0, &lev);
+
+    Serial.begin(BAUD_RATE);
+    Serial.println("Running");
+
+    // setup lev display
+    lev.onDataPage(levBaseDataPageHandler);
+    lev.onLevSpeedSystemInformation1(levSpeedSystemInformation1Handler);
+    lev.onLevSpeedDistanceInformation(levSpeedDistanceInformationHandler);
+    lev.onLevAltSpeedDistanceInformation(levAltSpeedDistanceInformationHandler);
+    lev.onLevSpeedSystemInformation2(levSpeedSystemInformation2Handler);
+    lev.onLevBatteryInfo(levBatteryInfo);
+    lev.onLevCapabilities(levCapabilities);
+    lev.onManufacturersInformation(manufacturersInformationDataPageHandler);
+    lev.onProductInformation(productInformationDataPageHandler);
+    lev.onChannelEvent(levAntChannelEvent);
+    lev.begin();
+    delay(100); // wait for module initialization
+
+    // wait for pair to complete
+    uint8_t status = lev.waitForPair();
+
+    // print channel status
+    Serial.println("===========================");
+    printStatus(status);
+    Serial.print("Device Number: ");
+    Serial.println(lev.getDeviceNumber());
+    Serial.print("Transmisison Type: ");
+    Serial.println(lev.getTransmissionType());
+}
+
+void loop() {
+    router.loop();
+}
+
+void levAntChannelEvent(ChannelEventResponse& msg, uintptr_t data) {
+    if (msg.getCode() == STATUS_EVENT_CHANNEL_CLOSED) {
+        Serial.println("channel closed - reconnect");
+        lev.begin();
+    }
+}
+
+void levBaseDataPageHandler(AntRxDataResponse& msg, uintptr_t data) {
+    LevBaseMainDataPage dp = LevBaseMainDataPage(msg);
+    Serial.println("===========================");
+    Serial.print("DataPage: ");
+    Serial.println(dp.getDataPageNumber());
+}
+
+void levSpeedSystemInformation1Handler(LevSpeedSystemInformation1& msg, uintptr_t data) {
+    Serial.print("Temperature state: ");
+    Serial.println(msg.getTemperatureState());      // TODO enums for temperature state
+    Serial.print("Travel mode state: ");
+    Serial.println(msg.getTravelModeState());       // TODO decode travel mode state
+    Serial.print("System state: ");
+    Serial.println(msg.getSystemState());           // TODO enums for system state
+    Serial.print("Gear state: ");
+    Serial.println(msg.getGearState());             // TODO decode gear state
+    Serial.print("Gear error: ");
+    Serial.println(msg.getErrorMessage());          // TODO enums for error message
+    Serial.print("Speed: ");
+    Serial.print(msg.getSpeed()/10);
```

Comment:
if I am correct, all these getSpeed calls are getting truncated due to integer division, / 10.0 should fix it, make sure to check the rest of the divisions :)

Reply:
There are two print statements, the first one for integer and the second one for the fraction. Of course you can use float calculation also, but this would be the only place in the whole program which uses floating point data types. I dont know exactly, what all the arduino toolchains will do, but they probably will link the floating point runtime library, due to this statement.

Feel free to change it in your branch.

Reply:
Yep you are right, i totally missed that.